_posts/2017-04-24-solarized.md

```diff
@@ -0,0 +1,545 @@
+```
+
+Here is an example of dircolors from Termite with 80% transparency:
+
+![dircolors solarized](/assets/images/dircolors-solarized.jpg)
+
+
+# [](#i3)i3
+
+> i3 is a tiling window manager, allowing one to always see all opened
+> windows, without them stacking above each other. With plenty of keystrokes,
+> it makes it possible to control every windows only using the keyboard, making
+> the mouse pretty much useless.
+
+My i3 stack consists of:
+
+* [i3-gaps-next](https://github.com/Airblader/i3), the
+  development branch of a fork of i3wm offering more features, like gaps
+  between windows instead of borders.
+* [i3blocks](https://github.com/Airblader/i3blocks-gaps) for the status line
+* [i3lock](https://i3wm.org/i3lock/) as screen locker.
+
+In addition, for my solarized theme switcher, I used
+a fork of [i3-style](https://github.com/acrisci/i3-style), where I defined my
+own transparent solarized themes, since I wasn't satisfied by the existing
+theme.
+
+To execute the script, I added a new binding in i3's configuration file:
+
+```config
+bindsym $mod+b  exec --no-startup-id ${HOME}/scripts/toggle_solarized_theme.sh
+```
+
+My `$mod` button is set to <kbd>Mod4</kbd>, which is the Windows key in my XPS
+13. Hence, whenever I press <kbd>Mod4</kbd>-<kbd>b</kbd>, my applications will
+toggle between dark and light solarized modes.
+Why using **b** as keystroke? Well, it stands for "Background", which color is
+the main difference between both modes of the solarized palette.
+
+Finally, I added the following contents to the script:
+
+```bash
+#i3-style solarized-transparent-${newmode} -o ${HOME}/.config/i3/config --reload
+coffee ${HOME}/scripts/cli.coffee solarized-transparent-${newmode} -o ${HOME}/.config/i3/config --reload
+```
+
+If my pull request gets accepted upstream, I'll then be able to use the normal
+i3-style script instead of the current coffee command.
+
+Here is how my i3 statusbar looks like:
+
+![i3blocks solarized](/assets/images/i3blocks-solarized.jpg)
+
+# [](#feh) feh
+
+> feh is an image viewer that can also be used to set the desktop background
+> image.
+
+To set the background image, simply pass the `--bg-fill` parameter to feh:
+
+```console
+$ feh --bg-fill "${HOME}.config/wallpapers/solarized-mountains-${newmode}.png"
+```
+
+That will also create a file `~/.fehbg`, which can then be reused.
+In my i3 configuration, I added the following line for that purpose:
+
+```
+exec_always --no-startup-id ~/.fehbg
+```
+
+For background image, I took a dark solarized image from [Deviant
+Art](http://9beat7.deviantart.com/art/Solarized-Mountains-530027093):
+
+![solarized background dark](/assets/images/solarized-mountains-dark.png)
+
+Then I modified it with [Gimp](https://www.gimp.org/) to get a light solarized version.
+You normally only need to change the 4 background/foreground colors when
+toggling mode (Accent colors staying the same in both modes). But in his image,
```

Comment:
> You normally only need to change the 4 background/foreground colors when
+toggling mode (Accent colors staying the same in both modes)

You normally only need to change the 4 background/foreground colors when toggling mode, accent colors staying the same in both modes.